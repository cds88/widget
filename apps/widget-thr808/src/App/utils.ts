import { isWidgetTHR08State } from '@widget/organisms';
import { fetchWidgetState } from '../utils/action';
import { WidgetTHR08State } from '@widget/organisms/';

export const fetchData = async (): Promise<WidgetTHR08State> => {
  const data = await fetchWidgetState();
  if (isWidgetTHR08State(data)) return data;
  throw new Error('Invalid data format');
};
