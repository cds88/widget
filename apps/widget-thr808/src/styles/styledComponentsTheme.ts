export const styledComponentsTheme = {
  styled: {
    colors: {
      indicator: {
        selected: 'rgb(153, 217, 234)',
        notSelected: 'rgb(8, 76, 148)',
      },
    },
  },
};
