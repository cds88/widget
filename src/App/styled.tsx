import { styled } from '@mui/system';
import Box from '@mui/material/Box';

export const AppWrapper = styled(Box)`
  display: flex;
  background: ${({theme})=>`${theme.palette.background.default}`};
  justify-content: center;
  align-items: center;
  height: 100vh;
`
