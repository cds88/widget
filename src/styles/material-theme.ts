import { createTheme } from '@mui/material';

export const appTheme = createTheme({
  typography: {
    htmlFontSize: 16,
    fontSize: 16,
    h1: {
      fontSize: '3rem',
    },
    h2: {
      fontSize: '2.5rem',
    },
    h3: {
      fontSize: '2rem',
    },
    h4: {
      fontSize: '1.75rem',
    },
    h5: {
      fontSize: '1.5rem',
    },
    h6: {
      fontSize: '1.25rem',
    },
    body1: {
      fontSize: '1rem',
    },

    button: {
      fontSize: '0.875rem',
    },
  },
  components: {
    MuiSwitch: {
      styleOverrides: {
        colorSecondary: {
          '&$checked': {
            color: 'linear-gradient(white, #f0f0f0)',
          },
        },
        switchBase: {
          '&.Mui-checked': {
            color: '#FFFFFF',
          },
        },
        track: {
          '&.Mui-checked': {
            backgroundColor: '#FFFFFF',
          },
          '$checked$checked + &': {
            opacity: 0.7,
            backgroundColor: '#fff',
          },
          backgroundColor: 'linear-gradient(white, #f0f0f0)',
        },
      },
    },
  },
  cssVariables: true,
  palette: {
    mode: 'dark',
    primary: {
      main: '#673AB7',
    },
    background: {
      default: 'rgba(3,21,45,255)',
      paper: 'rgba(16,41,72,255)',
    },
  },
});
