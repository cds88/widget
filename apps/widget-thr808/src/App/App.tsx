import { useSuspenseQuery } from '@tanstack/react-query';
import { useState } from 'react';

import { WidgetTHR08 } from '@widget/organisms';
import { AVAILABLE_WIDGETS } from '@widget/organisms/const';
import { WidgetTHR08State } from '@widget/organisms/WidgetTHR08/types';
import { fetchData } from './utils';

const App = () => {
  const { data } = useSuspenseQuery<WidgetTHR08State, Error>({
    queryKey: [AVAILABLE_WIDGETS.WIDGET_TH08],
    queryFn: fetchData,
  });

  const [widgetTHR08State, setWidgetTHR08State] =
    useState<WidgetTHR08State>(data);
  const { brightness, timeLeft, ...widgetTHR08ButtonPanels } = widgetTHR08State;

  const onSetBrightness = (value: number) => {
    setWidgetTHR08State((state) => ({
      ...state,
      brightness: value,
    }));
  };

  const onSetButtonPanel = (
    buttonPanelKeyValPair: Partial<
      Pick<WidgetTHR08State, 'flashing' | 'nightVision' | 'duskTillDawn'>
    >,
  ) => {
    setWidgetTHR08State((state) => ({
      ...state,
      ...buttonPanelKeyValPair,
    }));
  };

  return (
    <WidgetTHR08
      brightness={brightness}
      onSetBrightness={onSetBrightness}
      timeLeft={timeLeft}
      onSetButtonPanel={onSetButtonPanel}
      widgetTHR08ButtonPanels={widgetTHR08ButtonPanels}
    />
  );
};

export default App;
