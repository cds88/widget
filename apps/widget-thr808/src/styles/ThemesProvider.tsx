import { ThemeProvider as MaterialUIThemeProvider } from '@mui/material';
import { ThemeProvider as StyledComponentsThemeProvider } from 'styled-components';
import { materialUiTheme } from './materialTheme';
import { styledComponentsTheme } from './styledComponentsTheme';
export const ThemesProvider: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => (
  <MaterialUIThemeProvider theme={materialUiTheme}>
    <StyledComponentsThemeProvider theme={styledComponentsTheme}>
      {children}
    </StyledComponentsThemeProvider>
  </MaterialUIThemeProvider>
);
