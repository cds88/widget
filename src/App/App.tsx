import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import React, { Suspense } from 'react';
import { ErrorBoundary } from 'react-error-boundary';
import WidgetTHR08 from '../components/WidgetOrganisms/WidgetTHR08';
 
import { WidgetErrorFallback, WidgetLoadingFallback } from '../components/WidgetMolecules/WidgetFallbacks';
import { AppWrapper } from './styled';
const queryClient = new QueryClient();



const App = () => {
 
  return (
    <AppWrapper>
      <QueryClientProvider client={queryClient}>
        <Suspense fallback={<WidgetLoadingFallback  />}>
          <ErrorBoundary FallbackComponent={WidgetErrorFallback}>
            <WidgetTHR08 />
          </ErrorBoundary>
        </Suspense>
      </QueryClientProvider>
    </AppWrapper>
  );
};

export default App;
