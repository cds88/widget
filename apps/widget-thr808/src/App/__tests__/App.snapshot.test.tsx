import React from 'react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { render } from '@testing-library/react';
import {
  ThemeProvider as MaterialUIThemeProvider,
  CssBaseline,
} from '@mui/material';
import { ThemeProvider as StyledComponentsThemeProvider } from 'styled-components';
import App from '../App';
import { materialUiTheme } from '../../styles/materialTheme';
import { styledComponentsTheme } from '../../styles/styledComponentsTheme';

jest.mock('@widget/organisms/WidgetTHR08', () => () => (
  <div>Mocked WidgetTHR08</div>
));

const queryClient = new QueryClient();

describe('App Component Snapshot Test', () => {
  test('matches the snapshot', () => {
    const { asFragment } = render(
      <QueryClientProvider client={queryClient}>
        <MaterialUIThemeProvider theme={materialUiTheme}>
          <StyledComponentsThemeProvider theme={styledComponentsTheme}>
            <CssBaseline />
            <App />
          </StyledComponentsThemeProvider>
        </MaterialUIThemeProvider>
      </QueryClientProvider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
