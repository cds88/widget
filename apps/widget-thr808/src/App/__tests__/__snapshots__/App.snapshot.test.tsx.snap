// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Snapshot Test matches the snapshot 1`] = `<DocumentFragment />`;
